service/raft/raft_group_registry.cc

```diff
@@ -468,8 +471,24 @@ raft_server_with_timeouts::run_with_timeout(Op&& op, const char* op_name,
         if (!expiry.abort_source().abort_requested() || (as && as->abort_requested())) {
             throw;
         }
-        // TODO: improve error message (check for quorum loss etc)
-        const auto message = ::format("group [{}] raft operation [{}] timed out", _group_server.gid, op_name);
+        auto message = ::format("group [{}] raft operation [{}] timed out", _group_server.gid, op_name);
+        {
+            auto fd = _registry.failure_detector();
+            unsigned voters_count = 0;
+            unsigned alive_voters_count = 0;
+            for (const auto& c: _group_server.server->get_configuration().current) {
+                if (c.can_vote) {
+                    ++voters_count;
+                    if (fd->is_alive(c.addr.id)) {
+                        ++alive_voters_count;
+                    }
+                }
+            }
+            if (alive_voters_count <= voters_count / 2) {
+                message += ::format(", there is no raft quorum, total voters {}, alive voters {}",
+                    voters_count, alive_voters_count);
```

Comment:
It would be nice to report the set of alive (or dead) nodes as well.

Reply:
There could be a lot of them in big clusters. Now this message is thrown to the user, shouldn't contain hundreds of IPs. Maybe we can include several random down nodes for convenience.

Reply:
Having more than half of the nodes down in a huge cluster should not be a normal event. And eventually we will only have a handful of voters in such situations (we will maintain small subset of voters only. not each node will be a voter), so it we will report dead voters only the list should be relatively short.

Reply:
done